onnxruntime/test/providers/cpu/nn/pool_op_test.cc

```diff
@@ -203,6 +220,44 @@ TEST(PoolTest, MaxPool1D) {
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kCudaNHWCExecutionProvider, kTensorrtExecutionProvider});
 }
 
+TEST(PoolTest, MaxPool1D_case2) {
+  OpTester test("MaxPool");
+  // no padding
+  test.AddAttribute("auto_pad", "VALID");
+  test.AddAttribute("strides", std::vector<int64_t>{1});
+  test.AddAttribute("pads", vector<int64_t>{0, 0});
+  test.AddAttribute("kernel_shape", vector<int64_t>{2});
+
+  std::vector<float> x_vals = {1, 2, 3, 4, 5};
+  std::vector<int64_t> x_dims = {1, 1, 5};
+  // The last dim is (5-2+1)/1 = 4
+  std::vector<int64_t> expected_dims = {1, 1, 4};
+  std::vector<float> expected_vals = {2, 3, 4, 5};
+
+  test.AddInput<float>("X", x_dims, x_vals);
+  test.AddOutput<float>("Y", expected_dims, expected_vals);
+  test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kCudaNHWCExecutionProvider, kTensorrtExecutionProvider});
```

Comment:
Remove kCudaNHWCExecutionProvider here if we want to test NHWC.

Reply:
I think `kCudaNHWCExecutionProvider` is something else. The new ops are part of `kMsDomain` in the regular `kCudaExecutionProvider`. The tests for it (Pool op) are here - [onnxruntime/onnxruntime/test/providers/cuda/nhwc/pool_test.cc](https://github.com/microsoft/onnxruntime/blob/89aa4697b145b3879c8a0f40db353af1e5e5003a/onnxruntime/test/providers/cuda/nhwc/pool_test.cc#L53)

Reply:
When I removed it,  the test crashed. :-)